linker-private-library/src/main/scala/org/scalajs/linker/runtime/RuntimeLong.scala

```diff
@@ -566,21 +562,89 @@ object RuntimeLong {
       asUnsignedSafeDouble(lo, hi).toString
     } else {
       /* At this point, (lo, hi) >= 2^53.
-       * We divide (lo, hi) once by 10^9 and keep the remainder.
+       *
+       * The idea is to divide (lo, hi) once by 10^9 and keep the remainder.
        *
        * The remainder must then be < 10^9, and is therefore an int32.
        *
        * The quotient must be <= ULong.MaxValue / 10^9, which is < 2^53, and
        * is therefore a valid double. It must also be non-zero, since
        * (lo, hi) >= 2^53 > 10^9.
        *
-       * To avoid allocating a tuple with the quotient and remainder, we push
-       * the final conversion to string inside unsignedDivModHelper. According
-       * to micro-benchmarks, this optimization makes toString 25% faster in
-       * this branch.
+       * We should do that single division as a Long division. However, that is
+       * slow. We can cheat with a Double division instead.
+       *
+       * We convert the unsigned value num = (lo, hi) to a Double value
+       * approxNum. This is an approximation. It can lose as many as 11
+       * low-order bits. Hence |approxNum - num| <= 2^12.
+       *
+       * We then compute an approximated quotient
+       *   approxQuot = floor(approxNum / 10^9)
+       * instead of the theoretical value
+       *   quot = floor(num / 10^9)
+       *
+       * Since 10^9 > 2^29 > 2^12, we have |approxNum - num| < 10^9.
+       * Therefore, |approxQuot - quot| <= 1.
+       *
+       * That means that the approximated quotient is at most 1 unit away from
+       * the theoretical result.
```

Comment:
I would remove this paragraph. IMO "unit" can be confused with floating point unit. I anyways only states what the equation above states already.